test/TestArrayUtils.cpp

```diff
@@ -0,0 +1,1214 @@
+}
+
+GPU_TEST(array_utils, fill_one)
+{
+   int tempb[1] = {1};
+   // if you attempt to initialize this as b(tempb) without the size, the test fails. Maybe that constructor should be
+   // disabled for host_device pointer with cuda active?
+   care::host_device_ptr<int> b(tempb, 1, "fillhostdev");
+
+   care_utils::ArrayFill<int>(b, 1, -12);
+
+   EXPECT_EQ(b.pick(0), -12);
+}
+
+GPU_TEST(array_utils, fill_three) {
+   int tempb[3] = {1, 2, 3};
+   // if you attempt to initialize this as b(tempb) without the size, the test fails. Maybe that constructor should be
+   // disabled for host_device pointer with cuda active?
+   care::host_device_ptr<int> b(tempb, 3, "fillhostdev3");
+
+   care_utils::ArrayFill<int>(b, 3, -12);
+   EXPECT_EQ(b.pick(0), -12);
+   EXPECT_EQ(b.pick(1), -12);
+   EXPECT_EQ(b.pick(2), -12);
+}
+
+GPU_TEST(array_utils, min_empty)
+{
+  care::host_device_ptr<int> a;
+  // this works even when the start index is greater than length.
+  int initVal = -1;
+  int result = care_utils::ArrayMin<int>(a, 0, initVal, 567);
+  EXPECT_EQ(result, initVal);
+}
+
+GPU_TEST(array_utils, min_innerloop)
+{
+  // this tests the local_ptr version of min
+  int temp0[7] = {2, 1, 1, 8, 3, 5, 7};
+  int temp1[7] = {3, 1, 9, 10, 0, 12, 12};
+  care::host_device_ptr<int> ind0(temp0, 7, "mingpu0");
+  care::host_device_ptr<int> ind1(temp1, 7, "mingpu1");
+
+  care::host_device_ptr<int> a[2] = {ind0, ind1};
+
+  RAJAReduceMin<bool> passed{true};
+  LOOP_REDUCE(i, 0, 1) {
+     care::local_ptr<int> arr0 = a[0];
+     care::local_ptr<int> arr1 = a[1];
+
+     // min of entire array arr0
+     int result = care_utils::ArrayMin<int>(arr0, 7, 99, 0);
+     if (result != 1) {
+        passed.min(false);
+     }
+
+     // min of arr0 starting at index 6
+     result = care_utils::ArrayMin<int>(arr0, 7, 99, 6);
+     if (result != 7) {
+        passed.min(false);
+     }
+
+     // min of entire array arr1
+     result = care_utils::ArrayMin<int>(arr1, 7, 99, 0);
+     if (result != 0) {
+        passed.min(false);
+     }
+
+     // value min of arr1 with init val -1
+     result = care_utils::ArrayMin<int>(arr1, 7, -1, 7);
+     if (result != -1) {
+        passed.min(false);
+     }
+
+  } LOOP_REDUCE_END
+
+   ASSERT_TRUE((bool) passed);
+}
+
+
+GPU_TEST(array_utils, min_seven)
+{
+  int temp[7] = {2, 1, 1, 8, 3, 5, 7};
+  care::host_device_ptr<int> a(temp, 7, "minseven");
+  int initVal = 99;
+  // min of whole array
+  int result = care_utils::ArrayMin<int>(a, 7, initVal, 0);
+  EXPECT_EQ(result, 1);
+  
+  // min starting at index 3
+  result = care_utils::ArrayMin<int>(a, 7, initVal, 3);
+  EXPECT_EQ(result, 3);
+
+  // test init val -1
+  initVal = -1;
+  result = care_utils::ArrayMin<int>(a, 7, initVal, 0);
+  EXPECT_EQ(result, -1);
+}
+
+GPU_TEST(array_utils, max_empty)
+{
+  care::host_device_ptr<int> a;
+  // ArrayMin has a value for start index but ArrayMax does not. TODO: use slicing or pointer arithmatic instead.
```

Comment:
can you add a github ticket for this?

Reply:
Added two tickets:
1) For making the start index issue consistent, see https://rzlc.llnl.gov/jira/browse/ALE-2364 
2) More generally, for initial values when doing max, we sometimes use user-provided initial values, sometimes explicitly assume a double type, and sometimes (host only, standard library not availible on the device) use std::numeric_limits<T> . We should make this all consistent, see https://rzlc.llnl.gov/jira/browse/ALE-2365

Reply:
I made github tickets also. Those were jira tickets, I was confused